src/main/java/com/spikes2212/dashboard/AutoChooser.java

```diff
@@ -0,0 +1,144 @@
+package com.spikes2212.dashboard;
+
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+
+import java.util.HashSet;
+
+/**
+ * A class that implements {@link SendableChooser} in order to make an autonomous command chooser.
+ *
+ * @author Ofri Rosenbaum
+ * @see SendableChooser
+ */
+public class AutoChooser extends SendableChooser<Command> {
+
+    /**
+     * The default name that will be given to the {@link RootNamespace} in case no name was given in the constructor.
+     */
+    public static final String DEFAULT_NAMESPACE_NAME = "autonomous chooser";
+
+    /**
+     * The {@link RootNamespace} that this {@link SendableChooser} will be on.
+     */
+    protected RootNamespace rootNamespace;
+
+    /**
+     * A {@link HashSet} that contains the names of all the options in this {@link SendableChooser}.
+     */
+    private final HashSet<String> names = new HashSet<>();
+
+    /**
+     * Creates an autonomous chooser from the given commands, where the first command will be the default option and the
+     * rest of the commands will be the rest of the options.
+     *
+     * <p>All options will have names matching their command's {@link Class#getSimpleName()}. In case of multiple
+     * instances of the same class, numbers will also be added to differentiate between the names.</p>
+     *
+     * @param rootNamespaceName the name that will be given to the {@link RootNamespace}
+     * @param defaultOption the default command this {@link SendableChooser} will use as the default option
+     * @param options the commands that will be added to this {@link SendableChooser} other than the default command
+     */
+    public AutoChooser(String rootNamespaceName, Command defaultOption, Command... options) {
+        rootNamespace = new RootNamespace(rootNamespaceName);
+        String defaultName = defaultOption.getClass().getSimpleName();
+        addName(defaultName);
+        setDefaultOption(defaultName, defaultOption);
+        for (Command option : options) {
+            addOption(option.getClass().getSimpleName(), option);
+        }
+        putOnShuffleboard();
+    }
+
+    /**
+     * Creates an autonomous chooser from the given commands, where the first command will be the default option and the
+     * rest of the commands will be the rest of the options.
+     *
+     * <p>All options will have names matching their command's {@link Class#getSimpleName()}. In case of multiple
+     * instances of the same class, numbers will also be added to differentiate between the names.</p>
+     *
+     * @param defaultOption the default command this {@link SendableChooser} will use as the default option
+     * @param options the commands that will be added to this {@link SendableChooser} other than the default command
+     */
+    public AutoChooser(Command defaultOption, Command... options) {
+        this(DEFAULT_NAMESPACE_NAME, defaultOption, options);
+    }
+
```

Comment:
add a second constructor similar to the previous, using the default namespace name